specification/cognitiveservices/data-plane/AutoSuggest/stable/v1.0/AutoSuggest.json

```diff
@@ -22,11 +22,11 @@
       ]
     }
   },
-  "host": "api.cognitive.microsoft.com",
+  "host": "*.api.cognitive.microsoft.com",
```

Comment:
What is your intention of the preceding * in the host URL?

Reply:
Hi @jhendrixMSFT, this is just for avoid confusing.

Put * in there to make developer aware that this is just a placeholder.
If I just use "api.cognitive.microsoft.com", it looks like a real host, and the developer might assume that this is a working default value.

Reply:
I see.  So why didn't you mark all the methods as ```x-ms-parameterized-host``` instead?  Using this extension makes providing a value for the prefix mandatory.  With your current changes it makes this look like a legal default value.

Reply:
@jhendrixMSFT Because we need the capability to let users to customize endpoint.

They might host their own services in somewhere or their applications might be behind a reverse proxy. We want to allow them to use SDKs in those scenarios. x-ms-parameterized-host cannot help in that.

Reply:
So parameterize the entire URL and model the enums as ```string:true``` so callers can pass their own URLs but we still get codegen for the default ones you define in your enumeration.
```json
"x-ms-parameterized-host": {
  "hostTemplate": "{EndpointURL}",	
    "parameters": [	
    {	
      "$ref": "#/parameters/EndpointURL"	
    }	
  ]	
}
```
The problem with your current change is it hides the fact that the URL must be specified.  Here's a client ctor for Go.
```go
// New creates an instance of the BaseClient client.
func New() BaseClient {
	return NewWithBaseURI(DefaultBaseURI)
}
```
Where ```DefaultBaseURI``` is the invalid URL.  How does the caller know that this ctor is invalid?

Reply:
@jhendrixMSFT good point.
Let me try it and probably send a new PR very soon:)